sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -345,6 +418,56 @@ internal static IReadOnlyList<EntityRecognitionTasksItem> ConvertToEntityRecogni
             return collection;
         }
 
+        internal static IReadOnlyCollection<ExtractKeyPhrasesActionResult> ConvertToExtractKeyPhrasesActionResults(AnalyzeJobState jobState, IDictionary<string, int> idToIndexMap)
+        {
+            var collection = new List<ExtractKeyPhrasesActionResult>();
+
+            // TODO - Map Error.Target to ActionResult
+            //var errorToIndexMap = new Dictionary<string, Dictionary<int, TextAnalyticsError>>();
+
+            //foreach (TextAnalyticsErrorInternal error in jobState.Errors)
+            //{
+            //    // Sample target string - "#/tasks/keyPhraseExtractionTasks/0"
+            //    string target = error.Target;
+
+            //    string[] segments = target.Split('/');
+            //    if (segments.Length == 4)
+            //    {
+            //        errorToIndexMap.Add(segments[2], new Dictionary<int, TextAnalyticsError>())
+            //        return targetMap;
+            //    }
+
+            //}
+            foreach (KeyPhraseExtractionTasksItem task in jobState.Tasks.KeyPhraseExtractionTasks)
+            {
+                collection.Add(new ExtractKeyPhrasesActionResult(ConvertToExtractKeyPhrasesResultCollection(task.ResultsInternal, idToIndexMap), jobState.LastUpdateDateTime, jobState.Errors, task.ResultsInternal.Errors.Any()));
```

Comment:
Each task has a `LastUpdateDateTime`. Please use that instead of `jobState.LastUpdateDateTime` that is generic for the whole job.

Reply:
updated